Style guide

kfrn opened the pull request:
Addresses #429 

* Adds guidelines for contributing to readme (mostly nicked from @kieranjol :)
* Adds a PR template

kfrn committed:
Code style: headings according to nesting level

kfrn committed:
Docs: add new section, Guidelines for contributing

Content supplied by @kieranjol,
https://github.com/amiaopensource/ffmprovisr/issues/429

kfrn committed:
Code style: add PR template

retokromer commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,4 @@
+## Checklist
+
+* [ ] I've referred to the the [Guidelines for contributing](https://github.com/amiaopensource/ffmprovisr/blob/gh-pages/readme.md#guidelines-for-contributing)
+* [ ] New recipe has been added to `recipes.txt`
```
I am not sure about the second point. Would this not confuse a person making her/his first contribution?

I am happy to sync form time to time, as I did in the past. (There is a on-liner here, but I personally use a short Perl script to gather the data from GitHub, to generate the correct output for FFCommand_Engine and to install it directly at the right place.)

kieranjol replied:
Thanks for clarifying Reto - I'm happy to remove it. I wasn't really familiar with the recipes.txt protocol and saw a recent commit you made and so I added it in.

retokromer replied:
I could add here a new document explaining what is FFCommand_Engine, how to compile its last version and how to import ffmprovisr’s recipe into it.

kieranjol commented:
Hi @kfrn thanks so much - i love the structure here. Other folks might have other suggestions but I'm happy with it, and I take Reto's point about removing the recipes.txt point..

privatezero commented:
This looks good to me as well!

kfrn commented:
Cheers! I'll remove the part about `recipes.txt` when I'm home this evening :)

kfrn committed:
Docs/PR template: remove point about adding to recipes.txt

Not user-friendly for beginners - and Reto's on the case ;)

https://github.com/amiaopensource/ffmprovisr/pull/430#discussion_r433718981

kfrn committed:
Fix typos

kfrn commented:
Now updated :)

kieranjol reviewed (approved):
This is so wonderful, I LOVE the checklist! Next level! Thank you!

kfrn commented:
Thanks everyone! :)
